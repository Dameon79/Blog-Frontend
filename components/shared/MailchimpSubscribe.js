import { Component } from 'react'

class MailchimpSubscribe extends Component {
  render () {
    return (
      <div className="c-mc-signup">
        <form action="https://herokuapp.us19.list-manage.com/subscribe/post?u=3afc697b401bde3fce74181c1&amp;id=c21e2cc493" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" className="c-mc-signup__validate" target="_blank" noValidate>
          <input type="email" defaultValue="" name="EMAIL" className="c-mc-signup__email" placeholder="E-mail address" required></input>
          <div style={{position: 'absolute', left: -5000 + 'px'}} aria-hidden="true">
            <input type="text" name="b_3afc697b401bde3fce74181c1_c21e2cc493" tabIndex="-1" defaultValue=""></input>
          </div>
          <div className="c-mc-signup__clear">
            <input type="submit"  value="Subscribe" name="subscribe" id="mc-embedded-subscribe" className="c-mc-signup__button"></input>
          </div>
        </form>
      </div>
    );
  }
}

export default MailchimpSubscribe