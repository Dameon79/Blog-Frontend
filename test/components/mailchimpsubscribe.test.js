import MailchimpSubscribe from '../../components/shared/MailchimpSubscribe'

describe('<MailchimpSubscribe />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <MailchimpSubscribe />
    )

    expect(component).toMatchSnapshot()
  })
})