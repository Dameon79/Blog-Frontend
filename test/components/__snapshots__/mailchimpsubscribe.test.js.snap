// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MailchimpSubscribe /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MailchimpSubscribe />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        action="https://herokuapp.us19.list-manage.com/subscribe/post?u=3afc697b401bde3fce74181c1&id=c21e2cc493"
        className="c-mc-signup__validate"
        id="mc-embedded-subscribe-form"
        method="post"
        name="mc-embedded-subscribe-form"
        noValidate={true}
        target="_blank"
      >
        <input
          className="c-mc-signup__email"
          defaultValue=""
          name="EMAIL"
          placeholder="E-mail address"
          required={true}
          type="email"
        />
        <div
          aria-hidden="true"
          style={
            Object {
              "left": "-5000px",
              "position": "absolute",
            }
          }
        >
          <input
            defaultValue=""
            name="b_3afc697b401bde3fce74181c1_c21e2cc493"
            tabIndex="-1"
            type="text"
          />
        </div>
        <div
          className="c-mc-signup__clear"
        >
          <input
            className="c-mc-signup__button"
            defaultValue="Subscribe"
            id="mc-embedded-subscribe"
            name="subscribe"
            type="submit"
          />
        </div>
      </form>,
      "className": "c-mc-signup",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "https://herokuapp.us19.list-manage.com/subscribe/post?u=3afc697b401bde3fce74181c1&id=c21e2cc493",
        "children": Array [
          <input
            className="c-mc-signup__email"
            defaultValue=""
            name="EMAIL"
            placeholder="E-mail address"
            required={true}
            type="email"
          />,
          <div
            aria-hidden="true"
            style={
              Object {
                "left": "-5000px",
                "position": "absolute",
              }
            }
          >
            <input
              defaultValue=""
              name="b_3afc697b401bde3fce74181c1_c21e2cc493"
              tabIndex="-1"
              type="text"
            />
          </div>,
          <div
            className="c-mc-signup__clear"
          >
            <input
              className="c-mc-signup__button"
              defaultValue="Subscribe"
              id="mc-embedded-subscribe"
              name="subscribe"
              type="submit"
            />
          </div>,
        ],
        "className": "c-mc-signup__validate",
        "id": "mc-embedded-subscribe-form",
        "method": "post",
        "name": "mc-embedded-subscribe-form",
        "noValidate": true,
        "target": "_blank",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-mc-signup__email",
            "defaultValue": "",
            "name": "EMAIL",
            "placeholder": "E-mail address",
            "required": true,
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "children": <input
              defaultValue=""
              name="b_3afc697b401bde3fce74181c1_c21e2cc493"
              tabIndex="-1"
              type="text"
            />,
            "style": Object {
              "left": "-5000px",
              "position": "absolute",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "",
              "name": "b_3afc697b401bde3fce74181c1_c21e2cc493",
              "tabIndex": "-1",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="c-mc-signup__button"
              defaultValue="Subscribe"
              id="mc-embedded-subscribe"
              name="subscribe"
              type="submit"
            />,
            "className": "c-mc-signup__clear",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-mc-signup__button",
              "defaultValue": "Subscribe",
              "id": "mc-embedded-subscribe",
              "name": "subscribe",
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          action="https://herokuapp.us19.list-manage.com/subscribe/post?u=3afc697b401bde3fce74181c1&id=c21e2cc493"
          className="c-mc-signup__validate"
          id="mc-embedded-subscribe-form"
          method="post"
          name="mc-embedded-subscribe-form"
          noValidate={true}
          target="_blank"
        >
          <input
            className="c-mc-signup__email"
            defaultValue=""
            name="EMAIL"
            placeholder="E-mail address"
            required={true}
            type="email"
          />
          <div
            aria-hidden="true"
            style={
              Object {
                "left": "-5000px",
                "position": "absolute",
              }
            }
          >
            <input
              defaultValue=""
              name="b_3afc697b401bde3fce74181c1_c21e2cc493"
              tabIndex="-1"
              type="text"
            />
          </div>
          <div
            className="c-mc-signup__clear"
          >
            <input
              className="c-mc-signup__button"
              defaultValue="Subscribe"
              id="mc-embedded-subscribe"
              name="subscribe"
              type="submit"
            />
          </div>
        </form>,
        "className": "c-mc-signup",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "https://herokuapp.us19.list-manage.com/subscribe/post?u=3afc697b401bde3fce74181c1&id=c21e2cc493",
          "children": Array [
            <input
              className="c-mc-signup__email"
              defaultValue=""
              name="EMAIL"
              placeholder="E-mail address"
              required={true}
              type="email"
            />,
            <div
              aria-hidden="true"
              style={
                Object {
                  "left": "-5000px",
                  "position": "absolute",
                }
              }
            >
              <input
                defaultValue=""
                name="b_3afc697b401bde3fce74181c1_c21e2cc493"
                tabIndex="-1"
                type="text"
              />
            </div>,
            <div
              className="c-mc-signup__clear"
            >
              <input
                className="c-mc-signup__button"
                defaultValue="Subscribe"
                id="mc-embedded-subscribe"
                name="subscribe"
                type="submit"
              />
            </div>,
          ],
          "className": "c-mc-signup__validate",
          "id": "mc-embedded-subscribe-form",
          "method": "post",
          "name": "mc-embedded-subscribe-form",
          "noValidate": true,
          "target": "_blank",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-mc-signup__email",
              "defaultValue": "",
              "name": "EMAIL",
              "placeholder": "E-mail address",
              "required": true,
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "children": <input
                defaultValue=""
                name="b_3afc697b401bde3fce74181c1_c21e2cc493"
                tabIndex="-1"
                type="text"
              />,
              "style": Object {
                "left": "-5000px",
                "position": "absolute",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "defaultValue": "",
                "name": "b_3afc697b401bde3fce74181c1_c21e2cc493",
                "tabIndex": "-1",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="c-mc-signup__button"
                defaultValue="Subscribe"
                id="mc-embedded-subscribe"
                name="subscribe"
                type="submit"
              />,
              "className": "c-mc-signup__clear",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-mc-signup__button",
                "defaultValue": "Subscribe",
                "id": "mc-embedded-subscribe",
                "name": "subscribe",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
